### ISSUE TITLE
Yarn fails to initialize/run on script containing only commands (no dialogue)

### ISSUE THREAD

**Issue by fmoo** [YarnSpinnerTool/YarnSpinner-Unity#12]
**What is the current behavior?**

On Start, yarn throws the following unhandled exception:
```
UnassignedReferenceException: The variable baseLocalisationStringTable of YarnProgram has not been assigned.
You probably need to assign the baseLocalisationStringTable variable of the YarnProgram script in the inspector.
Yarn.Unity.DialogueRunner.AddStringTable (YarnProgram yarnScript) (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:201)
Yarn.Unity.DialogueRunner.CreateDialogueInstance () (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:584)
Yarn.Unity.DialogueRunner.get_Dialogue () (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:143)
Yarn.Unity.DialogueRunner.Start () (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:541)
```
When attempting to call NodeExists() the following unhandled exception is thrown, halting execution and preventing yarn from being used:
```
Cannot add a command handler for wait: one already exists
UnityEngine.Debug:LogError(Object)
Yarn.Unity.DialogueRunner:AddCommandHandler(String, BlockingCommandHandler) (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:358)
Yarn.Unity.DialogueRunner:CreateDialogueInstance() (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:581)
Yarn.Unity.DialogueRunner:get_Dialogue() (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:143)
Yarn.Unity.DialogueRunner:NodeExists(String) (at Library/PackageCache/dev.yarnspinner.unity@06a986f603/Runtime/DialogueRunner.cs:295)
...
```

**Please provide the steps to reproduce, and if possible a minimal demo of the problem**:
1. Create a .yarn file whose nodes only contain <<commands>>
2. Add the .yarn file to your Dialogue Runner's "Yarn Scripts"
3. Run

**What is the expected behavior?**
The program launches with no exceptions.  Scripts can be loaded and nodes in the script, executed.

**Please tell us about your environment:**
  - Yarn Spinner Version: master (github revision 06a986f603733a635fab88b5820fd9be9eaa2286)
  - Unity Version: 2019.3.15f1

**Other information** 
- workaround is to create a dummy node in my .yarn files with placeholder text that is never called, but this is pretty sloppy as I can't reuse node names and it wastes space.

**Comment by janmaslov** [comment]
Hi!

I've also noticed this with previous versions of Yarn Spinner, as I'm using Yarn to set up levels upon loading, and these scripts don't have dialogue. Easy, better, fix: Wrap two of the `AddStringTable()` calls in `DialogueRunner.cs` in an if-statement:

Around line 156, in the `Add(YarnProgram scriptToLoad)` method:
```
public void Add(YarnProgram scriptToLoad)
{
	Dialogue.AddProgram(scriptToLoad.GetProgram());
	AddStringTable(scriptToLoad);
}
```

Becomes:
```
public void Add(YarnProgram scriptToLoad)
{
	Dialogue.AddProgram(scriptToLoad.GetProgram());
	if(scriptToLoad.baseLocalisationStringTable != null){
		AddStringTable(scriptToLoad);
	}
}
```
---
And around line 583 (according to the current DialogueRunner.cs), in the `CreateDialogueInstance()` method:
```
foreach (var yarnScript in yarnScripts) {
	AddStringTable(yarnScript);
}
```

Becomes:
```
foreach (var yarnScript in yarnScripts) {
	if(yarnScript.baseLocalisationStringTable != null){
		AddStringTable(yarnScript);
	}
}
```

@desplesda, maybe there's a more robust way, or is this enough for normal operation? I didn't experience any problems doing it like this.

**Comment by fmoo** [comment]
> Easy, better, fix: Wrap two of the `AddStringTable()` calls in `DialogueRunner.cs` in an if-statement:

@janmaslov oh, awesome.  Think it's worth putting up a PR for?

**Comment by desplesda** [comment]
Another potential solution is to guarantee that `baseLocalisationStringTable` is never null, and starts with an empty dictionary?

**Comment by fmoo** [comment]
Hey all, any hope for a fix in the upcoming 2.0 release?

**Comment by desplesda** [comment]
This behaviour no longer exists as of version 2.0. As such, I'll close this issue here; thanks for filing the report!
